src/rgw/rgw_iam_policy.cc

```diff
@@ -1208,9 +1208,9 @@ ostream& operator <<(ostream& m, const Condition& c) {
   if (c.ifexists) {
     m << "IfExists";
   }
-  m << ": { " << c.key;
+  m << ": { " << c.key << ": ";
   print_array(m, c.vals.cbegin(), c.vals.cend());
-  return m << "}";
+  return m << " } }";
```

Comment:
@qrGitHub @adamemerson  i don't think we should quote a `Condition` element with parentheses. instead, we should quote the all the `Condition` elements with parentheses. see http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition.html. i am posting a fix at #18961. 

do it make sense to you?

Reply:
Ok then.